### PR TITLE
Add process to detect duplicate dependencies in webpack build

### PR DESCRIPTION
**Pull request by zephraph**
Currently it's really hard to know when there are duplicate dependencies in our webpack build. And we have a _lot_ of duplicates. This is part 1 of a solution to (hopefully) resolve that problem (or at least provide visibility around it)

This adds a webpack plugin called `inspectpack` which will generate a duplication report in the `.artifacts` directory. This will eventually be used to generate a comment in a PR that adds (or removes) duplicate bundles.

### CODE REVIEW

File: webpack/envs/base.js [hunk 1]
```diff
@@ -70,6 +73,24 @@ export const baseConfig = {
       jade: "jade/runtime.js",
       waypoints: "jquery-waypoints/waypoints.js",
     }),
+    new DuplicatesPlugin({
+      verbose: true,
+      emitHandler(report) {
+        const artifactsPath = path.join(process.cwd(), ".artifacts")
```
Comment (damassi) [inline webpack/envs/base.js]: `process.cwd()` -> `basePath`

File: webpack/envs/base.js [hunk 2]
```diff
@@ -70,6 +73,24 @@ export const baseConfig = {
...
+          .mkdir(artifactsPath)
+          .catch(() => Promise.resolve()) // .artifact directory exists, continue...
+          .then(() =>
+            fs.writeFile(
+              path.join(artifactsPath, "duplicates-report"),
+              stripAnsi(report)
+            )
+          )
+          .catch(err => {
+            console.error("Could not write duplicates report", err)
```
Comment (damassi) [inline webpack/envs/base.js]: console.error(`[DuplicatesPlugin] Could not write duplicates report, err`)

File: webpack/envs/base.js (hunk 1 above)
Comment (zephraph) [reply webpack/envs/base.js]: basePath?

File: webpack/envs/base.js (hunk 2 above)
Comment (zephraph) [reply webpack/envs/base.js]:
```suggestion
            console.error("[DuplicatesPlugin] Could not write duplicates report", err)
```

File: webpack/envs/base.js (hunk 1 above)
Comment (damassi) [reply webpack/envs/base.js]: From 
```
import { NODE_ENV, basePath, isCI } from "../../src/lib/environment"
```
up above

Comment (zephraph) [reply webpack/envs/base.js]:
```suggestion
        const artifactsPath = path.join(basePath, ".artifacts")
```

**Comment by damassi** [comment]
👍

**Pull request merged**